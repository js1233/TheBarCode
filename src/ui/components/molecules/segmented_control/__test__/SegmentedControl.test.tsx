import React from "react";
import {
  fireEvent,
  render,
  RenderAPI
} from "@testing-library/react-native";
import {
  Choice,
  SegmentedControl
} from "ui/components/molecules/segmented_control/SegmentedControl";

describe("Segmented control tests", () => {
  let onChangeMockedFn: jest.Mock<void, [Choice, number]>;
  let getByText: RenderAPI["getByText"];
  let toJSON: RenderAPI["toJSON"];
  let update: RenderAPI["update"];

  const values: Choice[] = [
    { label: "Label 1", value: "Value 1" },
    { label: "Label 2", value: "Value 2" },
    { label: "Label 3", value: "Value 3" }
  ];

  beforeEach(() => {
    onChangeMockedFn = jest.fn();
    ({ getByText, toJSON, update } = render(
      <SegmentedControl
        values={values}
        onChange={onChangeMockedFn}
        selectedIndex={1}
      />
    ));
  });

  test("snapshot", () => {
    expect(toJSON()).toMatchSnapshot();
  });

  test("contains values provided", () => {
    expect(getByText(values[0].label)).not.toBeNull();
    expect(getByText(values[1].label)).not.toBeNull();
    expect(getByText(values[2].label)).not.toBeNull();
  });

  test("callback fires when a tab is clicked", () => {
    // when
    fireEvent.press(getByText(values[2].label));

    // then
    expect(onChangeMockedFn).toBeCalledWith(values[2], 2);
  });

  test("throw error when values length lesser than  2 is passed", (doneCallback) => {
    // given
    const emptyList: Choice[] = [];
    try {
      update(
        <SegmentedControl
          values={emptyList}
          onChange={onChangeMockedFn}
          selectedIndex={1}
        />
      );
    } catch (e: any) {
      // then
      expect(e.message).toBeDefined();
      doneCallback();
    }
  });
});
