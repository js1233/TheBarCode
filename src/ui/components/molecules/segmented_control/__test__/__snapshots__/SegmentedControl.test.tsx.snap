// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Segmented control tests snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "elevation": 3,
      "paddingHorizontal": 18,
      "paddingVertical": 10,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.4,
      "shadowRadius": 2,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 10,
          "borderStyle": "solid",
          "flexDirection": "row",
          "height": 35,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#E5E5E5",
        },
        undefined,
      ]
    }
  >
    <View
      collapsable={false}
      nativeID="animatedComponent"
      style={
        Object {
          "backgroundColor": "#0F766E",
          "borderRadius": 10,
          "bottom": 0,
          "end": 0,
          "position": "absolute",
          "start": 0,
          "top": 0,
          "transform": Array [
            Object {
              "translateX": 0,
            },
          ],
          "width": 0,
        }
      }
      testID="animatedView"
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "ProximaNova-Bold",
              },
              Object {
                "color": "#171717",
                "fontSize": 27,
              },
              Object {
                "lineHeight": 35,
              },
              Array [
                Object {
                  "fontSize": 25,
                },
                Array [
                  Object {
                    "color": "#737373",
                  },
                ],
              ],
            ]
          }
        >
          Label 1
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "ProximaNova-Bold",
              },
              Object {
                "color": "#171717",
                "fontSize": 27,
              },
              Object {
                "lineHeight": 35,
              },
              Array [
                Object {
                  "fontSize": 25,
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                  },
                ],
              ],
            ]
          }
        >
          Label 2
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "flex": 1,
            "opacity": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "ProximaNova-Bold",
              },
              Object {
                "color": "#171717",
                "fontSize": 27,
              },
              Object {
                "lineHeight": 35,
              },
              Array [
                Object {
                  "fontSize": 25,
                },
                Array [
                  Object {
                    "color": "#737373",
                  },
                ],
              ],
            ]
          }
        >
          Label 3
        </Text>
      </View>
    </View>
  </View>
</View>
`;
